src/transformStateWithClones.js

```diff
@@ -7,7 +7,37 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const clone = [{ ...state }];
+  let count = 0;
+
+  for (const action of actions) {
+    const { type, extraData, keysToRemove } = action;
+
+    if (count > 0) {
+      clone.push({ ...clone[count - 1] });
+    }
```

Comment:
you don't need this if statement if you just move this operation down, after the switch/case